app/interactors/creates_company/create.rb

```diff
@@ -0,0 +1,23 @@
+module CreatesCompany
+  class Create
+    include Interactor
+
+    delegate :company_params, to: :context
+
+    def call
+      context.company = company
+
+      raise_error if company.invalid?
+    end
+
+    private
+
+    def company
+      @company ||= Company.create(company_params)
+    end
+
+    def raise_error
+      context.fail!(error: I18n.t("flash.users.company.failure"))
+    end
+  end
+end
```

Comment:
1. Could you explain what advantages interactors like this (Companies) give, please?
2. In this case (1 interactor for creating and updating), aren't we violating SRP?
3. Could you tell me, please, is it good practice to call interactors are verbs?

Reply:
1. In my opinion, `CreatesCompany::Create` a little confusing because the `create` word is repeated twice. Also, we have two similar names `CreateCompany` and `CreatesCompany::Create`. Also, the `creates` word in the 3d form (`creates`, not `create`). We have to use verb in the 1st form. 
2. No, we are not violating SRP principle because we have an interactor that does one specific thing — saves a company. And its name confirms that. 
3. Yes, it's a good practice. `CreateCompany` is also a verb. `CompanyCreation` is a noun and it is a bad name for an interactor.
You can read more here https://www.notion.so/flatstack/c144540b0cee40e8af8985a218282304
> Называйте интеракторы глаголом, или словосочетанием с глаголом. Это не принципиально, но в нашей компании так принято. Например CreateUser а не UserCreation и не UserCreator.

Reply:
2. I think I got your idea overnight. Answering my own question, we will have another interactor, where we will process the company parameters as we need and send them to the `Companies::Save` interactor?

Reply:
1. Yes, sure. You can leave it that way if you like it more. It's not a problem at all :)
2. Yes, as an option. Depends on the requirements that will be in the future.